chat/message/src/MessageLinks/MessageLinks.spec.tsx

```diff
@@ -0,0 +1,334 @@
+        });
+      });
+    });
+  });
+
+  describe('callback handling', () => {
+    test('calls onLinkClick when link is clicked', async () => {
+      const mockOnLinkClick = jest.fn();
+
+      renderMessageLinks({ onLinkClick: mockOnLinkClick });
+
+      const toggleButton = screen.getByRole('button');
+      await userEvent.click(toggleButton);
+
+      // Wait for expansion and then click a link
+      await waitFor(() => {
+        expect(toggleButton).toHaveAttribute('aria-expanded', 'true');
+      });
+
+      const firstLink = screen.getByText('Link 1');
+      await userEvent.click(firstLink);
+
+      expect(mockOnLinkClick).toHaveBeenCalled();
+    });
+  });
+
+  describe('keyboard navigation', () => {
+    test('toggle button is focusable', () => {
+      renderMessageLinks();
+      const toggleButton = screen.getByRole('button');
+
+      toggleButton.focus();
+      expect(toggleButton).toHaveFocus();
+    });
+
+    test('can be activated with Enter key', async () => {
+      renderMessageLinks();
+
+      const toggleButton = screen.getByRole('button');
+      toggleButton.focus();
+
+      await userEvent.keyboard('{Enter}');
+
+      expect(toggleButton).toHaveAttribute('aria-expanded', 'true');
+    });
+
+    test('can be activated with Space key', async () => {
+      renderMessageLinks();
+
+      const toggleButton = screen.getByRole('button');
+      toggleButton.focus();
+
+      await userEvent.keyboard(' ');
+
+      expect(toggleButton).toHaveAttribute('aria-expanded', 'true');
+    });
+  });
+
+  describe('edge cases', () => {
+    test('handles empty links array', () => {
+      renderMessageLinks({ links: [] });
+
+      expect(screen.getByText('Related Resources')).toBeInTheDocument();
+      expect(screen.getByRole('button')).toBeInTheDocument();
+      expect(screen.getByTestId('rich-links-area')).toBeInTheDocument();
+    });
+
+    test('handles missing onLinkClick prop', async () => {
+      renderMessageLinks({ onLinkClick: undefined });
+
+      const toggleButton = screen.getByRole('button');
+      await userEvent.click(toggleButton);
+
+      // Should still expand without errors
+      await waitFor(() => {
+        expect(toggleButton).toHaveAttribute('aria-expanded', 'true');
+      });
+    });
+
+    test('updates height when links change', async () => {
+      const { rerender } = renderMessageLinks();
+
+      const newLinks = [
+        ...mockLinks,
+        { children: 'Link 4', href: 'https://example.com/4' },
+        { children: 'Link 5', href: 'https://example.com/5' },
+      ];
+
+      rerender(
+        <LeafyGreenChatProvider variant={Variant.Compact}>
+          <MessageLinks {...defaultProps} links={newLinks} />
+        </LeafyGreenChatProvider>,
+      );
+
+      // Should still work with new links
+      expect(screen.getByText('Link 4')).toBeInTheDocument();
+      expect(screen.getByText('Link 5')).toBeInTheDocument();
+    });
+  });
+
+  describe('dark mode', () => {
```

Comment:
Do we need this test? It's the same as the rendering tests above, but only the darkMode prop is different.